### PR TITLE
EMSUSD-277 fix lost muted anonymous layers

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
- Add a global functionality to hold onto muted layers.
- This is to work around the fact that OpenUSD will not keep muted layers in memory.
- Make sure to clean up the list of muted layers when a new Maya scene is created.
- Use the muted layer holder in the MuteLayer command.
- The base class of the command was trying to hold onto layer, but when the undo stack is cleared, so would be the anonymous layers.
- In particular, when unloading a Maya Reference, Maya always clears the undo stack.
- Add a test that check if muted anonymous layers are lost when editing a Maya Reference.
- The test fails without my fix, works with it.
- Improved test helper functions to enable creating anonymous sub-layers.

### CODE REVIEW

File: lib/mayaUsd/commands/layerEditorCommand.cpp [hunk 1]
```diff
@@ -620,7 +621,7 @@ class MuteLayer : public BaseCmd
         // we perfer not holding to pointers needlessly, but we need to hold on to the layer if we
         // mute it otherwise usd will let go of it and its modifications, and any dirty children
         // will also be lost
```
Comment (samuelliu-adsk) [inline lib/mayaUsd/commands/layerEditorCommand.cpp:623]: I dont think we need this comment anymore

File: lib/mayaUsd/commands/layerEditorCommand.cpp [hunk 2]
```diff
@@ -638,8 +639,9 @@ class MuteLayer : public BaseCmd
             saveSelection();
             stage->MuteLayer(layer->GetIdentifier());
         }
+
         // we can release the pointer
-        _mutedLayer = nullptr;
+        removeMutedLayer(layer);
```
Comment (samuelliu-adsk) [inline lib/mayaUsd/commands/layerEditorCommand.cpp:644]: Question here: In AddMutedLayer, we recursively added all dirty sublayers to the mutedLayer list, so when remvoeMutedLayer should we do the same? For now it just removes one layer.

Comment (pierrebai-adsk) [reply lib/mayaUsd/commands/layerEditorCommand.cpp:644]: You're right, I will look into it.

File: lib/mayaUsd/commands/layerEditorCommand.cpp (hunk 1 above)
Comment (pierrebai-adsk) [reply lib/mayaUsd/commands/layerEditorCommand.cpp:623]: Actually, the comment says exactly why we are calling addMutedLayer, so I'll keep the comment.

**Comment by pierrebai-adsk** [comment]
Only failure is a crash on Linux in an interactive test about display layers, in malloc. Seems entirely unrelated.

**Review by samuelliu-adsk — APPROVED** [review]

**Pull request merged**